app/screens/ProfileScreen.tsx

```diff
@@ -20,7 +25,29 @@ export const ProfileScreen: FC<ProfileScreenProps> = function ProfileScreen(_pro
   return (
     <Screen preset="scroll" contentContainerStyle={$styles.barContainer} safeAreaEdges={["top"]}>
       <BackHeader navigation={navigation} title="Profile" onSavePress={handleSavePress} />
-      <BodyText variant="body4">{profile ? JSON.stringify(profile) : "Loading..."}</BodyText>
+      {isLoading && (
+        <YStack alignItems="center" justifyContent="center" py="$6" gap="$3">
+          <Spinner size="large" />
+          <BodyText variant="body" opacity={0.7}>
+            Loading your profile...
+          </BodyText>
+        </YStack>
+      )}
+
+      {!isLoading && error && (
+        <YStack px="$3" py="$4">
+          <BodyText variant="body">Couldn’t load your profile</BodyText>
+          <BodyText variant="body2" opacity={0.7}>
+            {error ? error.message : "Please try again."}
+          </BodyText>
+        </YStack>
+      )}
```

Comment:
also shouldn't need to optional check the error since we're already verifying the error exists with the logical AND

```suggestion
      {!isLoading && error && (
        <YStack px="$3" py="$4">
          <BodyText variant="body">Couldn’t load your profile</BodyText>
          <BodyText variant="body2" opacity={0.7}>
            {error.message}
          </BodyText>
        </YStack>
      )}
```